lib/castle/createCastleName.ts

```diff
@@ -0,0 +1,48 @@
+
+import { getPronoun } from '../npc-generation/gender/getPronoun'
+import { Castle } from '../../src/Castle/createCastle'
+import { Town } from '../town/_common'
+import { Namesake } from '../npc-generation/_common'
+
+export function createCastleName (town: Town, castle: Castle, namesake?: Namesake) {
+  console.log('Creating castle name...')
+
+  namesake = lib.createNamesake(town, namesake)
+  console.log(namesake)
+  const {
+    unique,
+    wordNouns,
+    nouns,
+    adjectives,
+    morphemes
+  } = lib.castleNames
+
+  const uniqueName = lib.random(unique)
+  const wordNoun = lib.random([lib.random(wordNouns), castle.wordNoun])
+  const noun = lib.random(nouns)
+  const adjective = lib.random(adjectives)
+  const prefix = lib.random(morphemes.prefix)
+  const suffix = lib.random(morphemes.suffix)
```

Comment:
We shouldn't be using `lib.<name>` for the files inside lib itself. Prefer importing these via a relative path instead.

Reply:
I'm unclear when importing from `lib` is "good practice"- should we turn it into an ESlint warning?

Reply:
We might be able to enforce it with typescript my removing access to the global value when it's under the `lib` folder.

But the gist is that the `lib` object is the final bundled `*.js`-file exposed namespace of that folder. Since everything under `src` is baked into the HTML, that code needs to use the exposed `lib` object instead of importing the files from the folder.